test/export/test_export.py

```diff
@@ -511,6 +510,23 @@ def method1(self, x: torch.Tensor) -> torch.Tensor:
 
         self.assertTrue(torch.allclose(eager_results, exported_results))
 
+    def test_export_constraint_discovery(self) -> None:
+        class RangeDiscovery(torch.nn.Module):
+            def forward(self, x):
+                if x.shape[0] > 4:
+                    return x + 1
+                else:
+                    return x - 1
+
+        m = RangeDiscovery()
+        x = torch.rand(5, 3)
+        constraints = [
+            dynamic_dim(x, 0) > 5,
+            dynamic_dim(x, 1),
+        ]
+        exported = export(m, (x,), constraints)
+        new_inp = torch.rand(6, 3)
+        self.assertEqual(exported(new_inp), RangeDiscovery()(new_inp))
```

Comment:
Should we also check that the graph only contains the add op?

Reply:
yeah sure.